packages/react-components/src/components/Badge/Badge.tsx

```diff
@@ -3,22 +3,33 @@ import * as React from 'react';
 import styles from './Badge.module.scss';
 import cx from 'clsx';
 
+const baseClass = 'badge';
+
 export interface BadgeProps extends React.HTMLAttributes<HTMLDivElement> {
-  secondary?: boolean;
+  kind?: 'primary' | 'secondary' | 'tertiary';
+  size?: 'large' | 'medium' | 'compact';
+  type?: 'content' | 'alert' | 'dot';
 }
 
 export const Badge: React.FC<BadgeProps> = ({
   children,
   className,
-  secondary = false,
+  kind = 'primary',
+  size = 'medium',
+  type = 'content',
 }) => {
-  return (
-    <span
-      className={cx(styles.badge, className, {
-        [styles['badge--secondary']]: secondary,
-      })}
-    >
-      {children}
-    </span>
+  const mergedClassNames = cx(
+    className,
+    styles[baseClass],
+    styles[`${baseClass}--${kind}`],
+    styles[`${baseClass}--${size}`]
   );
+
+  const content = {
+    ['content']: children,
```

Comment:
I know that we want to give the possibility to put our own content here, but I wonder if we should not make sure that too much content does not stretch this badge.
<img width="105" alt="badge-size" src="https://user-images.githubusercontent.com/23378800/180963185-85dad9bd-566c-4a89-acb4-94e5018ec246.png">

Reply:
Figma designs provide such examples 
<img width="91" alt="image" src="https://user-images.githubusercontent.com/8076673/180966015-bb7a8141-1406-48e4-b83c-811a25c56819.png">
so I believe `Badge` should be `stretchable`. The question is if we want to support dedicated handling for `Number` and automatically trim above 99 with `99+`.